### PR TITLE
add support for list data of top level. #31

### LINKED ISSUE #31: Need support for list data.
Example :
```
[  
   {
        "date": "2017-01-01",
        "title": "Automated question answering system using ontology and semantic role",
        "doc_id": "258cc0fd7973def42234f3db5ffdfd2a",
    },
    {
        "date": "2011-01-01",
        "title": "A new fuzzy-based method to weigh the related concepts in semantic focused web crawlers",
        "doc_id": "71ae01c53f4eeb7d402f5bcfbce36f9f",
     }
]
```

### PR DESCRIPTION
**Pull request by Ankush-Chander**
Hi @AtsushiSakai,

Thanks for this awesome project.

I was facing error while trying to load list of objects from json file.
I added a fix for same.

Please review.

### CODE REVIEW

File: pyjsonviewer/pyjsonviewer.py [hunk 1]
```diff
@@ -164,6 +164,7 @@ def find(self, search_text):
         self.collapse_all(None)
         for item_id in self.get_all_children(self.tree):
             item_text = self.tree.item(item_id, 'text')
+            item_text = str(item_text)
```
Comment (AtsushiSakai) [inline pyjsonviewer/pyjsonviewer.py:167]: why this change is needed?

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Thank you for great enhancement!!. 
Could you please add the json elements into the `sample.json` for testing.?
```
[  
   {
        "date": "2017-01-01",
        "title": "Automated question answering system using ontology and semantic role",
        "doc_id": "258cc0fd7973def42234f3db5ffdfd2a",
    },
    {
        "date": "2011-01-01",
        "title": "A new fuzzy-based method to weigh the related concepts in semantic focused web crawlers",
        "doc_id": "71ae01c53f4eeb7d402f5bcfbce36f9f",
     }
]
```

File: pyjsonviewer/pyjsonviewer.py (hunk 1 above)
Comment (Ankush-Chander) [reply pyjsonviewer/pyjsonviewer.py:167]: Thanks for prompt review.

I had some numeric values in my data. Search was breaking while applying `.lower()` on numeric data.
So I am converting all fields into `str`.
I am adding test cases for both the changes.

**Comment by Ankush-Chander** [comment]
Hey @AtsushiSakai,
I added test cases for the changes.
- [x] Add dat folder to store various sample files
- [x] Add dictionary test case
- [x] Add list test case
- [x] Add test case for search

You can run test as follows.
```
cd tests
pytest .
```

New dependency : pytest

Please review.

File: dat/dictionary.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (AtsushiSakai) [inline dat/dictionary.json:1]: Is it impossible to merge these json files into `sample.json`? I think it would be useful for manual testing.

File: tests/test_pyjsonviewer.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
```
Comment (AtsushiSakai) [inline tests/test_pyjsonviewer.py:1]: Could you please add a test for `sample.json`?

**Review by AtsushiSakai — COMMENTED** [review]
Thank you for adding a unit test. I have some comments.

File: tests/test_pyjsonviewer.py (hunk 3 above)
Comment (Ankush-Chander) [reply tests/test_pyjsonviewer.py:1]: I copied `sample.json`  into `dat/dictionary.json` so `test_dictionary_file` covers `sample.json`. So I think we can remove `sample.json` if it is not being used anywhere else.

File: dat/dictionary.json (hunk 2 above)
Comment (Ankush-Chander) [reply dat/dictionary.json:1]: We can merge the files into a valid `sample.json` but then sample.json would correspond to a nested object. It would be difficult to test `lists` and `objects` separately while keeping them in a single file.

File: tests/test_pyjsonviewer.py (hunk 3 above)
Comment (AtsushiSakai) [reply tests/test_pyjsonviewer.py:1]: Ah. Make sense. So, can you merge dictionary.json and list.json? I think list.json can be a part of a json element of dictionary.json. Is it impossible?

Comment (Ankush-Chander) [reply tests/test_pyjsonviewer.py:1]: You are right. I can add another test file `merged.json` which will have list as a field in dictionary object. Then I can add corresponding test case for those kind of files.

I hope that is what you are suggesting.

Comment (AtsushiSakai) [reply tests/test_pyjsonviewer.py:1]: Thanks. I want to keep a test json file is a one file, `sample.json`. I want to put all kinds of supports json representation into a single json file. That helps to explain this tool functionalities. So, I think maybe `dat` directory is not needed anymore.

Comment (Ankush-Chander) [reply tests/test_pyjsonviewer.py:1]: Actually code was already working for nested objects. `sample.json` has a fields `GlossSeeAlso` etc which are lists.
But it wasnt working for json files _which were lists to begin with._ 
That"s why I had to add seperate files `dictionary.json` which was an object and `list.json` which is a list.

By keeping them in same file we wont be able to test these cases seperately.

Comment (AtsushiSakai) [reply tests/test_pyjsonviewer.py:1]: > But it wasnt working for json files which were lists to begin with.
> That"s why I had to add seperate files dictionary.json which was an object and list.json which is a list.

Oh. Make sense. I understood why you are using two files.

**Review by AtsushiSakai — COMMENTED** [review]
When a list is at the top element, it will be expanded as trees.
SS 2021-05-04 at 22 12 48
But, it is at under second layers, it will be just as string.
SS 2021-05-04 at 22 12 11

It is not consistent. What do you think?

**Comment by Ankush-Chander** [comment]
Yes. It"s an existing behaviour. We are converting internal list nodes into string in `insert_node`

```
if type(value) is not dict:
            if type(value) is list:
                value = value[0:MAX_N_SHOW_ITEM]
                value = "[" + ",".join(map(str, value)) + "]"
            self.tree.insert(node, 'end', text=value, open=False)
```

I intend to make it recursive in upcoming pull request after testing.

**Review by AtsushiSakai — APPROVED** [review]
> I intend to make it recursive in upcoming pull request after testing.

Great!! I will create an issue for it.

LGTM for this PR.
